internal/cluster/cluster_test.go

```diff
@@ -0,0 +1,92 @@
+package cluster
+
+import (
+	"bytes"
+	"io"
+	"net/http"
+	"strings"
+	"testing"
+
+	apiTypes "github.com/konstructio/kubefirst-api/pkg/types"
+)
+
+type mockHTTPClient struct {
```

Comment:
Not something that needs change but don't forget there's a whole `httptest` package that adds support for mocking stuff out of the box (including http requests and responses). 

See here for reference: https://github.com/konstructio/dropkick/blob/main/internal/civo/sdk/json/json_fetch_test.go